src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/geoMap.js

```diff
@@ -158,6 +161,20 @@ class GeoMapVisualization extends Visualization {
     }
 
     onDataReceived(data) {
+        if (!this.isInit) {
+            this.setPreprocessor(
+                'df ->\n' +
+                    "    columns = df.select ['label', 'latitude', 'longitude'] . columns\n" +
+                    "    serialized = columns.map (c -> ['df_' + c.name, c.to_vector])\n" +
+                    '    Json.from_pairs serialized . to_text'
```

Comment:
Right now if the expression fails, we get the original data. Once we want to support multiple data types we can use `case` to transform depending on the input type.

Reply:
Ok, but that is an obscure feature I would say, and I feel it could change easily. So I would put here the case for Table explicitly and make `df.to_json.to_text` in other cases.

Reply:
Is refactored.